packages/web-components/fast-foundation/src/tree-view/tree-view.ts

```diff
@@ -88,24 +114,91 @@ export class TreeView extends FoundationElement {
             case keyHome:
                 if (this.treeItems && this.treeItems.length) {
                     TreeItem.focusItem(this.treeItems[0]);
+                    this.treeItems[0].setAttribute("tabindex", "0");
                 }
                 break;
             case keyEnd:
                 if (this.treeItems && this.treeItems.length) {
                     TreeItem.focusItem(this.treeItems[this.treeItems.length - 1]);
+                    this.treeItems[this.treeItems.length - 1].setAttribute(
+                        "tabindex",
+                        "0"
+                    );
+                }
+                break;
+            case keyArrowLeft:
+                e.preventDefault();
+                if (e.target && this.isFocusableElement(e.target as HTMLElement)) {
+                    const item = e.target as HTMLElement;
+                    if (item instanceof TreeItem && item.childItemLength() > 0) {
+                        item.expanded = false;
+                    }
+                }
+                break;
+            case keyArrowRight:
+                e.preventDefault();
+                if (e.target && this.isFocusableElement(e.target as HTMLElement)) {
+                    const item = e.target as HTMLElement;
+                    if (item instanceof TreeItem && item.childItemLength() > 0) {
+                        item.expanded = true;
+                    }
                 }
                 break;
+            case keyArrowDown:
+                e.preventDefault();
+                if (e.target && this.isFocusableElement(e.target as HTMLElement)) {
+                    this.focusNextNode(1, e.target as HTMLElement);
+                }
+                break;
+            case keyArrowUp:
+                e.preventDefault();
+                if (e.target && this.isFocusableElement(e.target as HTMLElement)) {
+                    this.focusNextNode(-1, e.target as HTMLElement);
+                }
+                break;
+            case keyEnter:
+                // In single-select trees where selection does not follow focus (see note below),
+                // the default action is typically to select the focused node.
+                this.handleSelected(e.target as HTMLElement);
+                break;
             default:
                 return true;
         }
     };
 
+    private focusNextNode(delta: number, item: HTMLElement): void {
```

Comment:
this method doesn't seem to do anything if `item` is an `HTMLElement`. Should you just expect a `TreeItem` instead?